lang/src/main/kotlin/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -390,6 +438,103 @@ private class BlobExprValue(private val ion: IonSystem, val value: ByteArray) :
     override val ionValue by lazy { toIonValue(ion) }
 }
 
+/**
+ * Core [ExprValue] over an [IonValue].
+ */
+internal class IonExprValue(private val valueFactory: ExprValueFactory, override val ionValue: IonValue) : BaseExprValue() {
```

Comment:
This internal was removed in https://github.com/partiql/partiql-lang-kotlin/pull/881/files, but now I decided to add it back. Since at that moment, the implementation of `ExprValueFactory.newFromIonValue(ionValue: IonValue)` was set to depend on `IonValue.toExprValue()`, and `IonValue.toExprValue()` uses factory pattern to create new `ExprValue`. However, in this PR, we changed `IonValue.toExprValue()` to use the new `ExprValue` construction methods. As a result, we need to add this class back so `ExprValueFactory.newFromIonValue(ionValue: IonValue)` method still use factory pattern to construct new `ExprValue`.  

To summarize, the old API is only being deprecated in this PR, not removed. So it should keep the same.